### PR TITLE
Introduce working e2e test for GH Actions [WATCHER]

### PR DESCRIPTION
**Pull request by khlifi411**
**Description**

Changes proposed in this pull request:

- Introduce e2e test as pre- and post-PR job (we first decided just to have it as post-job, but since it runs as long as the smoke tests run, we can also have it as pre-job); Used already existing go e2e test suite, and fixed it
- Job uses PR-Image in pre-merge run, and `latest` image after PR has been merged
- Upgrade setup-go to v4 in smoke tests (incl. bug fixes + remove `cache=true` since this is default in v4)
- Renamed `test.yaml` to `smoke-test.yaml` to be a bit more specific
- Renamed jobs, since names like `Test` are not specific enough
- Set kyma-reque-success-interval in test setup to 30seconds for easier debugging
- Fix local test setup guide
- Fix linting in `flags.go`
- Include more logging in e2e tests (`By`-steps)

**Related issue(s)**
#288

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+          fi
+          echo "/etc/hosts file patched"
+      - name: Install homebrew and Prerequisites
+        run: |
+            /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+            (echo; echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"') >> /root/.profile
+            eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+            sudo apt-get update -y
+            sudo apt-get install build-essential -y
+            mv brew /usr/local/bin
```
Comment (ruanxin) [inline .github/workflows/e2e-test.yml:34]: This brew is under `/home/linuxbrew/.linuxbrew/bin/brew`

**Review by lindnerby — DISMISSED** [review]

File: flags.go [hunk 2]
```diff
@@ -49,7 +49,6 @@ func defineFlagVar() *FlagVar {
 		defaultMaxConcurrentManifestReconciles,
 		"The maximum number of concurrent Manifest Reconciles which can be run.")
 	flag.IntVar(&flagVar.maxConcurrentWatcherReconciles, "max-concurrent-watcher-reconciles",
-		defaultMaxConcurrentWatcherReconciles,
 		"The maximum number of concurrent Watcher Reconciles which can be run.")
```
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply flags.go]: was merge conflicts, already reverted

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -15,5 +15,5 @@ configMapGenerator:
 
 images:
 - name: controller
-  newName: europe-docker.pkg.dev/kyma-project/prod/lifecycle-manager
```
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply config/manager/kustomization.yaml:18]: oh damn XD

**Pull request closed without merging**